File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int solve(vector<int>& nums, int index) {
        if (index >= nums.size() - 1) {
            return 0;
        }
        int count = INT_MAX - 1;
        for (int i = 1; i <= nums[index]; i++) {
            count = min(count, 1 + solve(nums, index + i));
        }
        return count;
    }
    
    int jump(vector<int>& nums) {
        return solve(nums, 0);
    }
};

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int lastJump = 0, maxReach = 0, steps = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            maxReach = Math.max(maxReach, i + nums[i]);
            if (i == lastJump) {
                lastJump = maxReach;
                steps++;
            }
        }
        return steps;
    }
}